lib/rebaser-core/src/api_types.rs

```diff
@@ -0,0 +1,483 @@
+pub mod enqueue_updates_request;
+pub mod enqueue_updates_response;
+
+use serde::{de::DeserializeOwned, Serialize};
+use strum::VariantNames;
+use thiserror::Error;
+
+const CONTENT_TYPE_CBOR: &str = "application/cbor";
+const CONTENT_TYPE_JSON: &str = "application/json";
+
+/// Alias for a type-erased error type.
+pub type BoxError = Box<dyn std::error::Error + Send + Sync>;
+
+#[derive(Debug, Error)]
+#[error("error serializing message: {0}")]
+pub struct SerializeError(#[source] BoxError);
+
+impl SerializeError {
+    pub fn from_err<E>(err: E) -> Self
+    where
+        E: std::error::Error + Send + Sync + 'static,
+    {
+        Self(Box::new(err))
+    }
+}
+
+#[derive(Debug, Error)]
+#[error("error derializing message: {0}")]
+pub struct DeserializeError(#[source] BoxError);
+
+impl DeserializeError {
+    pub fn from_err<E>(err: E) -> Self
+    where
+        E: std::error::Error + Send + Sync + 'static,
+    {
+        Self(Box::new(err))
+    }
+}
+
+#[derive(Debug, Error)]
+#[error("error upgrading message: {0}")]
+pub struct UpgradeError(#[source] BoxError);
+
+impl UpgradeError {
+    pub fn from_err<E>(err: E) -> Self
+    where
+        E: std::error::Error + Send + Sync + 'static,
+    {
+        Self(Box::new(err))
+    }
+}
+
+// TODO(fnichol): Guess what? We should consolidate these `id!` macros! Here's the current catch:
+// most of our "event" ids require serializing for over the wire transmission but don't always need
+// to write into the Postgres database. A lot of our `id!` macros out of dal (and now si-events)
+// get coupled to requiring Postgres marshalling so we just tack that on. For the moment this id
+// type has no Postgres requirements and so wil purposfully *omit* the `postgres_types::FromSql`
+// trait implementations. It think there's a way to split these concners up, but it requires a
+// couple more small crates and I'm just not down to do that right this moment. Cheers!
```

Comment:
Yeah, we need to tackle this at some point. They're bigger than they need to be and we have a lot of duplicated `id!` usage for semantically the same types. I'd borderline stuff it all in an `si-id` crate. We can talk about it later, though. Good call out!

Reply:
Yep, I almost made such a crate!